### ISSUE TITLE
Breaking from 2 to 3 (array and comments)

### ISSUE THREAD

**Issue by mfouilleul** [mikefarah/yq#364]
I have a breaking change in the version 3 with array and comment, can you help debug this, maybe there is a flag which can help me.

The file 1.txt
```
# any class that implements the SeedProvider interface and has a
# constructor that takes a Map<String, String> of parameters will do.
seed_provider:
    # Addresses of hosts that are deemed contact points.
    # Cassandra nodes use this list of hosts to find each other and learn
    # the topology of the ring.  You must change this if you are running
    # multiple nodes!
    - class_name: org.apache.cassandra.locator.SimpleSeedProvider
      parameters:
          # seeds is actually a comma-delimited list of addresses.
          # Ex: "<ip1>,<ip2>,<ip3>"
          - seeds: "127.0.0.1"
```

The file 2.txt
```
authenticator: PasswordAuthenticator
authorizer: org.apache.cassandra.auth.CassandraAuthorizer
```

I want to merge the both, 

With the version 2, the result is:
```
docker run --rm -it  mikefarah/yq:2 sh
/workdir # yq m 1.txt 2.txt
authenticator: PasswordAuthenticator
authorizer: org.apache.cassandra.auth.CassandraAuthorizer
seed_provider:
- class_name: org.apache.cassandra.locator.SimpleSeedProvider
  parameters:
  - seeds: 127.0.0.1
```

With the version 3, the result is:
```
# yq m 1.txt 2.txt
# any class that implements the SeedProvider interface and has a
# constructor that takes a Map<String, String> of parameters will do.
seed_provider:
- # Addresses of hosts that are deemed contact points.
  # Cassandra nodes use this list of hosts to find each other and learn
  # the topology of the ring.  You must change this if you are running
  # multiple nodes!
  class_name: org.apache.cassandra.locator.SimpleSeedProvider
  parameters:
  - # seeds is actually a comma-delimited list of addresses.
    # Ex: "<ip1>,<ip2>,<ip3>"
    seeds: "127.0.0.1"
```

My issue is the integration of the comment inside the array:
```
- # seeds is actually a comma-delimited list of addresses.
    # Ex: "<ip1>,<ip2>,<ip3>"
    seeds: "127.0.0.1"
```

The format `- seeds:` is necessary in the official cassandra's docker-entrypoint. 

Thanks!

**Comment by mfouilleul** [comment]
Btw, I agree that the generated YAML is valid.

**Comment by mikefarah** [comment]
I see - this is how the underlying yaml parser library encodes the yaml (https://godoc.org/gopkg.in/yaml.v3). It's annoying it doesn't keep the comments in the right place. I'd raise an issue and link it here (there may already be one?)

Like you say, the yaml is valid (and should be equivalant) - but I'm guessing cassandra doesn't like it?

Failing a yaml.v3 fix - the only thing I can think of is adding a strip comments flag that will remove all the comments...

**Comment by mikefarah** [comment]
Yep found an existing issue here:

https://github.com/go-yaml/yaml/issues/459

**Comment by mikefarah** [comment]
There is now a strip comments feature in yq to circumvent this issue:

https://mikefarah.gitbook.io/yq/usage/output-format#strip-comments

**Comment by mikefarah** [comment]
This has been fixed in 3.3.2
